dateutil/rrule.py

```diff
@@ -658,6 +659,23 @@ def __init__(self, freq, dtstart=None,
             self._bysecond = tuple(sorted(self._bysecond))
             self._original_rule['bysecond'] = self._bysecond
 
+        # rscale and optional skip values
+        if rscale is not None:
+            assert rscale == 'GREGORIAN', 'unsupported RSCALE'
+
+            self._original_rule['rscale'] = rscale
+
+            if skip is not None:
+                self._original_rule['skip'] = skip
+                self._skip = skip
+            else:
+                self._skip = 'OMIT'
+
+            assert self._skip in ['OMIT', 'BACKWARD', 'FORWARD'], 'invalid SKIP'
```

Comment:
I think in this case the possible values for `SKIP` are probably bounded such that an enum-like structure (can't use real enums because we still support Python 2.7) would be better here. In that case, I'd find an assertion of type or value more acceptable since it's probably a superfluous check that can be omitted in optimized code.